src/plugins/data/common/datasets/types.ts

```diff
@@ -247,6 +247,13 @@ export interface Dataset extends BaseDataset {
   timeFieldName?: string;
   /** Optional language to default to from the language selector */
   language?: string;
+  /** Optional reference to the source dataset. Example usage is for indexed views to store the
+   * reference to the table dataset
+   */
+  sourceDatasetRef?: {
```

Comment:
nit:
where you able to comment on @virajsanghvi previous feedback?
def could be follow up but just some insight url length is limited:
- Application and global state are stored in URL parameters
- Although we're far from URL length limits, field name length matters
- Should follow existing naming conventions (e.g. 'timeFieldName' from index patterns)
- Consider shorter property names when possible

Reply:
> Do you have a suggested name that accurately reflects what this is?

nope :). i didn't have a good suggestion and was being pedantic and something to keep track of.